plasmapy/simulation/particle_tracker.py

```diff
@@ -0,0 +1,945 @@
+            if self.field_weighting == "volume averaged":
+                _Ex, _Ey, _Ez, _Bx, _By, _Bz = grid.volume_averaged_interpolator(
+                    pos_tracked * u.m,
+                    "E_x",
+                    "E_y",
+                    "E_z",
+                    "B_x",
+                    "B_y",
+                    "B_z",
+                    persistent=True,
+                )
+            elif self.field_weighting == "nearest neighbor":
+                _Ex, _Ey, _Ez, _Bx, _By, _Bz = grid.nearest_neighbor_interpolator(
+                    pos_tracked * u.m,
+                    "E_x",
+                    "E_y",
+                    "E_z",
+                    "B_x",
+                    "B_y",
+                    "B_z",
+                    persistent=True,
+                )
+
+            # Interpret any NaN values (points off the grid) as zero
+            # Do this before adding to the totals, because 0 + nan = nan
+            _Ex = np.nan_to_num(_Ex, nan=0.0 * u.V / u.m)
+            _Ey = np.nan_to_num(_Ey, nan=0.0 * u.V / u.m)
+            _Ez = np.nan_to_num(_Ez, nan=0.0 * u.V / u.m)
+            _Bx = np.nan_to_num(_Bx, nan=0.0 * u.T)
+            _By = np.nan_to_num(_By, nan=0.0 * u.T)
+            _Bz = np.nan_to_num(_Bz, nan=0.0 * u.T)
+
+            # Add the values interpolated for this grid to the totals
+            Ex += _Ex
+            Ey += _Ey
+            Ez += _Ez
+            Bx += _Bx
+            By += _By
+            Bz += _Bz
+
+        # Create arrays of E and B as required by push algorithm
+        E = np.array(
+            [Ex.to(u.V / u.m).value, Ey.to(u.V / u.m).value, Ez.to(u.V / u.m).value]
+        )
+        E = np.moveaxis(E, 0, -1)
+        B = np.array([Bx.to(u.T).value, By.to(u.T).value, Bz.to(u.T).value])
+        B = np.moveaxis(B, 0, -1)
+
+        # Calculate the adaptive time step from the fields currently experienced
+        # by the particles
+        # If user sets dt explicitly, that's handled in _adaptive_dt
+        dt = self._adaptive_dt(Ex, Ey, Ez, Bx, By, Bz)
+
+        # TODO: Test v/c and implement relativistic Boris push when required
+        # vc = np.max(v)/_c
+
+        # Make sure the time step can be multiplied by a [nparticles, 3] shape field array
+        if dt.size > 1:
+            dt = dt[tracked_mask, np.newaxis]
+
+            # Increment the tracked particles' time by dt
+            self.time[tracked_mask] += dt
+        else:
+            self.time += dt
+
+        self.x[tracked_mask], self.v[tracked_mask] = boris_push(
+            pos_tracked, vel_tracked, B, E, self.q, self.m, dt, inplace=False
+        )
+
+        self.dt = dt
+
+    @property
+    def on_any_grid(self):
+        """
+        Binary array for each particle indicating whether it is currently
+        on ANY grid.
+        """
+        return np.sum(self.on_grid, axis=-1) > 0
+
+    @property
+    def vmax(self):
+        """
+        Calculate the maximum velocity.
```

Comment:
This first sentence needs to be a more complete description, because it shows up by itself in the docs in the parameter table:

https://plasmapy--2245.org.readthedocs.build/en/2245/api/plasmapy.simulation.particle_tracker.ParticleTracker.html#plasmapy.simulation.particle_tracker.ParticleTracker

How about "The maximum velocity of any particle in the simulation"